src/extension-client/remix-extension.ts

```diff
@@ -16,8 +20,20 @@ export class RemixExtension<T extends Api = any> {
   protected currentRequest: PluginRequest
   public isLoaded = false
 
-  constructor() {
+  constructor(opts: RemixExtensionOptions) {
     window.addEventListener('message', event => this.getMessage(event), false)
+
+    if (!opts.useCustomBootStrapTheme) this.initListenOnSwitchTheme()
+  }
+
+  private initListenOnSwitchTheme () {
```

Comment:
To avoid having to many `if` statement in the constructor we can have the `if` in the method : 
```typescript
private initListenOnSwitchTheme (hasCustomTheme: boolean) {
  if (hasCustomTheme) return
  //...
}
```